### PR TITLE
Clean up `handlers.js`

### PR DESCRIPTION
**Pull request by michaelgeorgeattard**

### CODE REVIEW

File: src/main/handlers.js [hunk 1]
```diff
@@ -7,26 +7,24 @@ const config = require('../config')
...
-  }
-  if (process.platform === 'linux') {
-    installLinux()
+  switch (process.platform) {
+    case 'darwin': installDarwin()
+      break
+    case 'win32': installWin32()
+      break
+    case 'linux': installLinux()
+      break
```
Comment (feross) [inline src/main/handlers.js]: Not a fan of `switch` statements. I'll take a small set of `if`s any day. Switches are prone to misbehaving if you forget to `break`.

File: src/main/handlers.js [hunk 2]
```diff
@@ -7,26 +7,24 @@ const config = require('../config')
...
-  }
-  if (process.platform === 'linux') {
-    uninstallLinux()
+  switch (process.platform) {
+    case 'darwin': uninstallDarwin()
+      break
+    case 'win32': uninstallWin32()
+      break
+    case 'linux': uninstallLinux()
+      break
```
Comment (feross) [inline src/main/handlers.js]: Same.

File: src/main/handlers.js [hunk 3]
```diff
@@ -42,9 +40,9 @@ function installDarwin () {
   // File handlers are defined in `Info.plist`.
 }
 
-function uninstallDarwin () {}
+function uninstallDarwin () { }
```
Comment (feross) [inline src/main/handlers.js]: Not a fan of random whitespace changes.

File: src/main/handlers.js [hunk 4]
```diff
@@ -42,9 +40,9 @@ function installDarwin () {
...
 
-const EXEC_COMMAND = [ process.execPath ]
+const EXEC_COMMAND = [process.execPath]
```
Comment (feross) [inline src/main/handlers.js]: Same.

File: src/main/handlers.js [hunk 5]
```diff
@@ -108,37 +106,37 @@ function installWin32 () {
...
     }
 
     function setIcon (err) {
-      if (err) log.error(err.message)
+      if (err) return log.error(err.message)
 
       const iconKey = new Registry({
         hive: Registry.HKCU,
-        key: '\\Software\\Classes\\' + protocol + '\\DefaultIcon'
+        key: `\\Software\\Classes\\${protocol}\\DefaultIcon`
```
Comment (feross) [inline src/main/handlers.js]: Have you tested this change? We don't have a test for this, so it should be manually tested before merging.

File: src/main/handlers.js (hunk 1 above)
Comment (Borewit) [reply src/main/handlers.js]: Not the most elegant syntax inherited from C, but it is intended for this scenario. From a logical point of view the elses are missing. Adding them with all the curly brackets is not a very pretty view neither. 

Lets get these pull requests out of the way and focus on important stuff.

File: src/main/handlers.js (hunk 3 above)
Comment (Borewit) [reply src/main/handlers.js]: Irrelevant indeed

**Comment by feross** [comment]
Thanks for the PR @michaelgeorgeattard! This has been superseded by https://github.com/webtorrent/webtorrent-desktop/pull/1406

**Comment by feross** [comment]
Re-opening, see: https://github.com/webtorrent/webtorrent-desktop/pull/1406#issuecomment-391973517

**Comment by michaelgeorgeattard** [comment]
Went through comments.

**Pull request merged**